opcua/server/subscription_service.py

```diff
@@ -17,86 +17,85 @@ def __init__(self, loop, aspace):
-                sub.publish([ack.SequenceNumber for ack in acks if ack.SubscriptionId == subid])
+        #with self._lock:
+        for subid, sub in self.subscriptions.items():
+            sub.publish([ack.SequenceNumber for ack in acks if ack.SubscriptionId == subid])
 
     def create_monitored_items(self, params):
         self.logger.info("create monitored items")
-        with self._lock:
-            if params.SubscriptionId not in self.subscriptions:
-                res = []
-                for _ in params.ItemsToCreate:
-                    response = ua.MonitoredItemCreateResult()
-                    response.StatusCode = ua.StatusCode(ua.StatusCodes.BadSubscriptionIdInvalid)
-                    res.append(response)
-                return res
-            return self.subscriptions[params.SubscriptionId].monitored_item_srv.create_monitored_items(params)
+        #with self._lock:
+        if params.SubscriptionId not in self.subscriptions:
+            res = []
+            for _ in params.ItemsToCreate:
+                response = ua.MonitoredItemCreateResult()
+                response.StatusCode = ua.StatusCode(ua.StatusCodes.BadSubscriptionIdInvalid)
+                res.append(response)
+            return res
+        return self.subscriptions[params.SubscriptionId].monitored_item_srv.create_monitored_items(params)
 
     def modify_monitored_items(self, params):
         self.logger.info("modify monitored items")
-        with self._lock:
-            if params.SubscriptionId not in self.subscriptions:
-                res = []
-                for _ in params.ItemsToModify:
-                    result = ua.MonitoredItemModifyResult()
-                    result.StatusCode = ua.StatusCode(ua.StatusCodes.BadSubscriptionIdInvalid)
-                    res.append(result)
-                return res
-            return self.subscriptions[params.SubscriptionId].monitored_item_srv.modify_monitored_items(params)
+        #with self._lock:
+        if params.SubscriptionId not in self.subscriptions:
+            res = []
+            for _ in params.ItemsToModify:
+                result = ua.MonitoredItemModifyResult()
+                result.StatusCode = ua.StatusCode(ua.StatusCodes.BadSubscriptionIdInvalid)
+                res.append(result)
+            return res
+        return self.subscriptions[params.SubscriptionId].monitored_item_srv.modify_monitored_items(params)
 
     def delete_monitored_items(self, params):
         self.logger.info("delete monitored items")
-        with self._lock:
-            if params.SubscriptionId not in self.subscriptions:
-                res = []
-                for _ in params.MonitoredItemIds:
-                    res.append(ua.StatusCode(ua.StatusCodes.BadSubscriptionIdInvalid))
-                return res
-            return self.subscriptions[params.SubscriptionId].monitored_item_srv.delete_monitored_items(
-                params.MonitoredItemIds)
+        #with self._lock:
```

Comment:
maybe remove these commented #with self._lock

Reply:
Thanks for reviewing! I will remove them once i had the time to check if they need to be replaced with a asynio.Lock

Reply:
In fact I reviewed quite a lot but could not find errors ;-) well done!